sim/rogue/ambush.go

```diff
@@ -42,21 +43,28 @@ func (rogue *Rogue) registerAmbushSpell() {
 			IgnoreHaste: true,
 		},
 		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+			if hasCutthroatRune && rogue.CutthroatProcAura.IsActive() && rogue.HasDagger(core.MainHand)	 {
+				return true
+			}
+			if hasCutthroatRune && rogue.HasDagger(core.MainHand) && rogue.IsStealthed() {
+				return true
+			}
 			return !rogue.PseudoStats.InFrontOfTarget && rogue.HasDagger(core.MainHand) && rogue.IsStealthed()
```

Comment:
Maybe we could consolidate this a bit
```suggestion
      if !rogue.HasDagger(core.MainHand) {
        return false
      }
			if hasCutthroatRune && (rogue.CutthroatProcAura.IsActive() || rogue.IsStealthed()) {
				return true
			}
			return !rogue.PseudoStats.InFrontOfTarget && rogue.IsStealthed()
```